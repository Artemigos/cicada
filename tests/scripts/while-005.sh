counter=17
while echo "$counter" | grep -iq "^1.$"
    break
done

while echo "$counter" | grep -iq "^1.$"
    if [ $counter = 18 ]
        counter=$(expr $counter + 1)
        break
    fi
    echo "counter = $counter"
    counter=$(expr $counter + 1)
done

counter=26
while echo "$counter" | grep -iq "^2.$"
    if [ $counter = 27 ]
        counter=$(expr $counter + 1)
        if [ $counter = 28 ]
            break
        fi
    fi
    echo "counter = $counter"
    counter=$(expr $counter + 1)
done

counter=36
while echo "$counter" | grep -iq "^3.$"
    if [ $counter = 37 ]
        counter=$(expr $counter + 1)
        if [ $counter = 38 ]
            counter=$(expr $counter + 1)
            break
        fi
    fi
    echo "counter = $counter"
    counter=$(expr $counter + 1)
done

counter=48
while echo "$counter" | grep -iq "^4.$"
    for x in foo bar
        if [ $x = bar ]
            break
        fi
        echo $x
    done
    echo "counter = $counter"
    counter=$(expr $counter + 1)
done
