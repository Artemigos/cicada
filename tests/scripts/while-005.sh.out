counter = 17
counter = 26
counter = 36
foo
counter = 48
foo
counter = 49
